controllers/velero_test.go

```diff
@@ -107,6 +112,160 @@ var (
 	}
 )
 
+type ReconcileVeleroControllerScenario struct {
+	namespace string
+	dpaName   string
+	envVar    corev1.EnvVar
+}
+
+var _ = ginkgo.Describe("Test ReconcileVeleroDeployment function", func() {
+	var (
+		ctx                 = context.Background()
+		currentTestScenario ReconcileVeleroControllerScenario
+		updateTestScenario  = func(scenario ReconcileVeleroControllerScenario) {
+			currentTestScenario = scenario
+		}
+	)
+
+	ginkgo.AfterEach(func() {
+		os.Unsetenv(currentTestScenario.envVar.Name)
+
+		deployment := &appsv1.Deployment{}
+		if k8sClient.Get(
+			ctx,
+			types.NamespacedName{
+				Name:      common.Velero,
+				Namespace: currentTestScenario.namespace,
+			},
+			deployment,
+		) == nil {
+			gomega.Expect(k8sClient.Delete(ctx, deployment)).To(gomega.Succeed())
+		}
+
+		dpa := &oadpv1alpha1.DataProtectionApplication{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      currentTestScenario.dpaName,
+				Namespace: currentTestScenario.namespace,
+			},
+		}
+		gomega.Expect(k8sClient.Delete(ctx, dpa)).To(gomega.Succeed())
+
+		namespace := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: currentTestScenario.namespace,
+			},
+		}
+		gomega.Expect(k8sClient.Delete(ctx, namespace)).To(gomega.Succeed())
+	})
+
+	ginkgo.DescribeTable("Check if Subscription Config environment variables are passed to Velero Containers",
+		func(scenario ReconcileVeleroControllerScenario) {
+			updateTestScenario(scenario)
+
+			namespace := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: scenario.namespace,
+				},
+			}
+			gomega.Expect(k8sClient.Create(ctx, namespace)).To(gomega.Succeed())
+
+			dpa := &oadpv1alpha1.DataProtectionApplication{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      scenario.dpaName,
+					Namespace: scenario.namespace,
+				},
+				Spec: oadpv1alpha1.DataProtectionApplicationSpec{
+					Configuration: &oadpv1alpha1.ApplicationConfig{
+						Velero: &oadpv1alpha1.VeleroConfig{
+							NoDefaultBackupLocation: true,
+						},
+					},
+				},
+			}
+			gomega.Expect(k8sClient.Create(ctx, dpa)).To(gomega.Succeed())
+
+			// Subscription Config environment variables are passed to controller-manager Pod
+			// https://github.com/operator-framework/operator-lifecycle-manager/blob/d8500d88932b17aa9b1853f0f26086f6ee6b35f9/doc/design/subscription-config.md
+			os.Setenv(scenario.envVar.Name, scenario.envVar.Value)
```

Comment:
Would be cool if this could undo the setenv

```
    originalValue, _ := os.LookupEnv("MY_VAR") // Store original value (if exists)
    defer func() {
        if originalValue != "" {
            os.Setenv("MY_VAR", originalValue) // Restore original value
        } else {
            os.Unsetenv("MY_VAR")             // Remove if newly created
        }
    }()

    os.Setenv("MY_VAR", "new_value")
```

Reply:
This was not as necessary in subscription suite test e2e because env is set in operator controller pod.

Now this is a unit test, setting env here could maybe change  current terminal env and break things like `git` which reads from env proxy settings.

Reply:
I think I already do this https://github.com/mateusoliveira43/oadp-operator/blob/fix/refactor-e2e-tests-more/controllers/velero_test.go#L131 https://github.com/mateusoliveira43/oadp-operator/blob/fix/refactor-e2e-tests-more/controllers/nodeagent_test.go#L47